File: Homeworks/C#2/Methods/06.FirstLargerThanNeighbours/FirstLargerThanNeighbours.cs
using System;
using System.Linq;

//Problem 6. First larger than neighbours

//Write a method that returns the index of the first element in array that is larger than its neighbours, or -1, if there’s no such element.
//Use the method from the previous exercise.

class FirstLargerThanNeighbours
{
    static void Main()
    {
        int n = int.Parse(Console.ReadLine());
        int[] array = new int[n];
        string textNumbers = Console.ReadLine();
        var numbers = textNumbers.Split(' ').ToArray();
        for (int i = 0; i < numbers.Length; i++)
        {
            array[i] = int.Parse(numbers[i]);
        }
        Console.WriteLine(FindLargerElement(array));
    }

    static int FindLargerElement(int[] array)
    {
        int count = 0;
        for (int position = 1; position < array.Length - 1; position++)
        {
            if (array[position] > array[position + 1] && array[position] > array[position - 1])
            {
                count++;
            }
            if (count == 1)
            {
                count = position;
                break;            
            }
        }
        return count;
    }

}

File: Homeworks/JSFundamentals/Fucntions/firstLargerThanNeighbours.js
var array = [2, 1, 3, 6, 9, 11, 15];


function checkElementAtPosition(array, position) {
    var i;
    if (position < 1 || position > array.length - 1) {
        return 'This element have not two neighbours';
    } else {
        i = position;
        if (array[i - 1] < array[i] && array[i] > array[i + 1]) {
            return true;
        } else {
            return false;
        }
    }
}

function firstLargerThanNeighbours(array) {
    for (var i = 1; i < array.length - 1; i++) {
        if (checkElementAtPosition(array, i) == true) {
            return i;
        }
    }
    return -1;
}


console.log(firstLargerThanNeighbours(array));